python/ray/data/_internal/remote_fn.py

```diff
@@ -24,6 +24,11 @@ def cached_remote_fn(fn: Any, **ray_remote_args) -> Any:
             # as needed.
             "scheduling_strategy": "DEFAULT",
             "max_retries": -1,
+            # Ray typically automatically retries system errors. However, in some cases,
+            # Ray won't retry system errors if they're raised from task code. To ensure
+            # Ray Data is fault tolerant to those errors, we need to add RaySystemError
+            # to the retry_exceptions list.
+            "retry_exceptions": [ray.exceptions.RaySystemError],
```

Comment:
what will happen if user already set this? If user already set it to a list, we should append this to it. If they already set it to `True`, we should do nothing.